### ISSUE TITLE
SAML2 IdP: creating callback service on startup causes audit failures

### ISSUE THREAD

**Issue by mmoayyed** [apereo/cas#2325]
Working with 5.0.2, adding saml-idp module and json-service registry. Embedded or otherwise. Latest java 8. 

When CAS attempts to create the callback service for the SAML IdP module, the auditing functionality bombs. 

```
Caused by: java.lang.IllegalArgumentException: 'clientIpAddress' cannot be null.
Check the correctness of @Audit annotation at the following audit point: execution(public synchronized org.apereo.cas.services.RegisteredService org.apereo.cas.services.DefaultServicesManagerImpl.save(org.apereo.cas.services.RegisteredService))
	at org.apereo.inspektr.audit.AuditActionContext.assertNotNull(AuditActionContext.java:80)
	at org.apereo.inspektr.audit.AuditActionContext.<init>(AuditActionContext.java:67)
	at org.apereo.inspektr.audit.AuditTrailManagementAspect.executeAuditCode(AuditTrailManagementAspect.java:159)
	at org.apereo.inspektr.audit.AuditTrailManagementAspect.handleAuditTrail(AuditTrailManagementAspect.java:147)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:629)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:618)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:70)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:168)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:656)
	at org.apereo.cas.services.DefaultServicesManagerImpl$$EnhancerBySpringCGLIB$$4289f6f4.save(<generated>)
	at org.apereo.cas.support.saml.web.idp.profile.AbstractSamlProfileHandlerController.registerCallback(AbstractSamlProfileHandlerController.java:214)
	at org.apereo.cas.support.saml.web.idp.profile.AbstractSamlProfileHandlerController.initialize(AbstractSamlProfileHandlerController.java:135)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleElement.invoke(InitDestroyAnnotationBeanPostProcessor.java:366)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeInitMethods(InitDestroyAnnotationBeanPostProcessor.java:311)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:134)
	... 39 more
```

**Comment by mmoayyed** [comment]
One temporary work around is to pre-create the callback service manually so CAS skips to add it.

**Comment by mmoayyed** [comment]
Fixed in 5.0.3
